LibAddonMenu-2.0/LibAddonMenu-2.0/controls/slider.lua

```diff
@@ -132,12 +137,23 @@ function LAMCreateControl.slider(parent, sliderData, controlName)
         control.slidervalueBG:SetDimensions(50, 16)
         control.slidervalueBG:SetAnchor(TOP, slider, BOTTOM, 0, 0)
     end
-    control.slidervalue = wm:CreateControlFromVirtual(nil, control.slidervalueBG, "ZO_DefaultEditForBackdrop")
-    local slidervalue = control.slidervalue
-    slidervalue:ClearAnchors()
-    slidervalue:SetAnchor(TOPLEFT, control.slidervalueBG, TOPLEFT, 3, 1)
-    slidervalue:SetAnchor(BOTTOMRIGHT, control.slidervalueBG, BOTTOMRIGHT, -3, -1)
-    slidervalue:SetTextType(TEXT_TYPE_NUMERIC)
+
+    local slidervalue
+    if(not sliderData.readOnly) then
+        control.slidervalue = wm:CreateControlFromVirtual(nil, control.slidervalueBG, "ZO_DefaultEditForBackdrop")
+        slidervalue = control.slidervalue
+        slidervalue:ClearAnchors()
+        slidervalue:SetAnchor(TOPLEFT, control.slidervalueBG, TOPLEFT, 3, 1)
+        slidervalue:SetAnchor(BOTTOMRIGHT, control.slidervalueBG, BOTTOMRIGHT, -3, -1)
+        slidervalue:SetTextType(TEXT_TYPE_NUMERIC)
+    else
+        control.slidervalue = wm:CreateControl(nil, control.slidervalueBG, CT_LABEL)
```

Comment:
Is it really necessary to create a label instead of an editbox? Can't you just call SetEditEnabled when it is read only?

Reply:
fixed.. tested all possibilities, should work fine